### ISSUE TITLE
qtservice

### ISSUE THREAD

**Issue by joeblew99** [therecipe/qt#252]
its seems that with QT you can create Services and Apps that communicate using QtRemoteObjects

https://www.kdab.com/qt-android-create-android-service-using-qt/

But when i looked for these objects in this code base i did not find any.

Can anyone advise on this ?

**Comment by bclermont** [comment]
just use a golang routine to run as an android service?

**Comment by joeblew99** [comment]
It's true i can write a normal golang apps that can run as service on desktop and even mobile.

But you need IPC between the service and the many apps you have on the device.

This was the original need because it allows a centralisation of common things ( crypto, key stores, talking to servers on an apps behalf even when all apps are closed).

On desktops i can use IPC and there are a few great libs that do this 100% cross platform on desktop.
But for mobile there is more complexity.

So i found this QT remote objects API and thought wow this is a very nice thing to give me IPC all desktops and mobiles for free. I can then of course out golang code inside it just like we do with a qml visual app.

Does this help explain my rationale ?

**Comment by therecipe** [comment]
>it might be only in 5.9, because this was the best description i could find. Very cool super powers with this :)

Yes, it will be available with 5.9 https://wiki.qt.io/New_Features_in_Qt_5.9 (as a preview module)
And the 5.9 beta should be available next month also, so you probably just need to wait 2-3 weeks to be able to test it.

**Comment by joeblew99** [comment]
i will leave this one open for now.
Will be good to test it when she ships.
